src/OpenTelemetry.Exporter.Geneva/Internal/ExporterEventSource.cs

```diff
@@ -43,40 +43,64 @@ public void FailedToSendTraceData(Exception ex)
+            this.FailedToSendTraceData(ToInvariantString(ex));
         }
     }
 
-    [Event(EVENT_ID_LOG, Message = "Exporter failed to send log data. Exception: {0}", Level = EventLevel.Error)]
+    [NonEvent]
     public void FailedToSendLogData(Exception ex)
     {
-        if (Log.IsEnabled(EventLevel.Error, EventKeywords.All))
+        if (this.IsEnabled(EventLevel.Error, EventKeywords.All))
         {
             // TODO: Do not hit ETW size limit even for external library exception stack.
-            this.WriteEvent(EVENT_ID_LOG, ToInvariantString(ex));
+            this.FailedToSendLogData(ToInvariantString(ex));
         }
     }
 
-    [Event(EVENT_ID_METRIC, Message = "Exporter failed to send metric data. Data will not be sent. MonitoringAccount = {0} MetricNamespace = {1}, MetricName = {2}, Message: {3}", Level = EventLevel.Error)]
+    [NonEvent]
     public void FailedToSendMetricData(string monitoringAccount, string metricNamespace, string metricName, Exception ex)
     {
-        if (Log.IsEnabled(EventLevel.Error, EventKeywords.All))
+        if (this.IsEnabled(EventLevel.Error, EventKeywords.All))
         {
             // TODO: Do not hit ETW size limit even for external library exception stack.
-            this.WriteEvent(EVENT_ID_METRIC, monitoringAccount, metricNamespace, metricName, ToInvariantString(ex));
+            this.FailedToSendMetricData(monitoringAccount, metricNamespace, metricName, ToInvariantString(ex));
         }
     }
 
-    [Event(EVENT_ID_ERROR, Message = "Exporter failed.", Level = EventLevel.Error)]
+    [NonEvent]
     public void ExporterException(string message, Exception ex)
     {
         if (Log.IsEnabled(EventLevel.Error, EventKeywords.All))
         {
             // TODO: Do not hit ETW size limit even for external library exception stack.
-            this.WriteEvent(EVENT_ID_ERROR, message, ToInvariantString(ex));
+            this.ExporterException(message, ToInvariantString(ex));
         }
     }
 
+    [Event(EVENT_ID_TRACE, Message = "Exporter failed to send trace data. Exception: {0}", Level = EventLevel.Error)]
+    public void FailedToSendTraceData(string error)
+    {
+        this.WriteEvent(EVENT_ID_TRACE, error);
+    }
+
+    [Event(EVENT_ID_LOG, Message = "Exporter failed to send log data. Exception: {0}", Level = EventLevel.Error)]
+    public void FailedToSendLogData(string error)
+    {
+        this.WriteEvent(EVENT_ID_LOG, error);
+    }
+
+    [Event(EVENT_ID_METRIC, Message = "Exporter failed to send metric data. Data will not be sent. MonitoringAccount = {0} MetricNamespace = {1}, MetricName = {2}, Message: {3}", Level = EventLevel.Error)]
+    public void FailedToSendMetricData(string monitoringAccount, string metricNamespace, string metricName, string error)
+    {
+        this.WriteEvent(EVENT_ID_METRIC, monitoringAccount, metricNamespace, metricName, error);
+    }
+
+    [Event(EVENT_ID_ERROR, Message = "Exporter failed.", Level = EventLevel.Error)]
```

Comment:
Probably should add the arguments in the message something like:

`Message = "Exporter failed. Message: {0}, Exception: {1}"`

Reply:
Updated.